src/NSwag.CodeGeneration.CSharp/Templates/Controller.liquid

```diff
@@ -91,8 +91,61 @@ public abstract class {{ Class }}ControllerBase : {% if HasBaseClass %}{{ BaseCl
     [System.Obsolete]
 {%         endif -%}
     [{{ AspNetNamespace }}.Http{{ operation.HttpMethodUpper }}, {{ AspNetNamespace }}.Route("{{ operation.Path }}"{% if operation.HasRouteName %}, Name = "{{ operation.RouteName }}"{% endif %})]
-    public abstract {%  if operation.WrapResponse %}System.Threading.Tasks.Task<System.Net.Http.HttpResponseMessage>{% else %}{{ operation.ResultType }}{% endif %} {{ operation.ActualOperationName }}({% for parameter in operation.Parameters %}{% if parameter.IsBody %}[{{ AspNetNamespace }}.FromBody] {% endif %}{{ parameter.Type }} {{ parameter.VariableName }}{% if GenerateOptionalParameters and parameter.IsOptional %} = null{% endif %}{% if parameter.IsLast == false or UseCancellationToken %}, {% endif %}{% endfor %}{% if UseCancellationToken %}System.Threading.CancellationToken cancellationToken{% endif %});
+    public abstract {%  if operation.WrapResponse %}System.Threading.Tasks.Task<System.Net.Http.HttpResponseMessage>{% else %}{{ operation.ResultType }}{% endif %} {{ operation.ActualOperationName }}({% for parameter in operation.Parameters %}{% if parameter.IsHeader %}[FromHeader{% if parameter.IsValidIdentifier == false %}(Name = "{{ parameter.Name }}"){% endif %}] {% endif %}{% if parameter.IsBody %}[{{ AspNetNamespace }}.FromBody] {% endif %}{{ parameter.Type }} {{ parameter.VariableName }}{% if GenerateOptionalParameters and parameter.IsOptional %} = null{% endif %}{% if parameter.IsLast == false or UseCancellationToken %}, {% endif %}{% endfor %}{% if UseCancellationToken %}System.Threading.CancellationToken cancellationToken{% endif %});
 
 {%     endfor -%}
 }
 {% endif -%}
+
+public class FromHeaderBinding : {{ AspNetNamespace }}.Controllers.HttpParameterBinding
```

Comment:
This should maybe only be generated in the aspnet and not aspnetcore case?

Reply:
Yes agreed.   I was thinking for the AspNetCore case, we could then conditionally insert this using statement (so the method declaration could then be the same in each case):

`using FromHeaderAttribute = Microsoft.AspNetCore.Mvc.FromHeaderAttribute;`

?

Reply:
_I've made the changes as above_ - that does mean it's a little inconsistent compared to how [FromBody] works (which has a fully-qualified namespace) - but it simplifies the template a bit where it's already quite complex.

Reply:
Same problem here: This class will be repeated... we need to add this as an additional code artifact in C# based on the selected option and target framework

Also I we really have to avoid usings as they pollute the current namespace (name clashes) and makes the code coupled to the file template (which is against the code artifact concept) - ideally the code should be self-describing and not depend on some usings somewhere else

Reply:
Ok no problem, I didn't appreciate that; we can the attributes self-describe.